src/service/alerts/alert.rs

```diff
@@ -539,14 +543,12 @@ pub async fn delete_by_id<C: ConnectionTrait>(
     let Some(alert) = db::alerts::alert::get_by_id(conn, org_id, alert_id).await? else {
         return Ok(());
     };
-
-    match db::alerts::alert::delete_by_id(conn, org_id, alert_id).await {
-        Ok(_) => {
-            remove_ownership(org_id, "alerts", Authz::new(&alert.name)).await;
-            Ok(())
-        }
-        Err(e) => Err(e.into()),
+    db::alerts::alert::delete_by_id(conn, org_id, alert_id).await?;
+    if let Some(id) = alert.id.map(|id| id.to_string()) {
+        remove_ownership(org_id, "alerts", Authz::new(&id)).await;
     }
+    remove_ownership(org_id, "alerts", Authz::new(&alert.name)).await;
```

Comment:
When an alert is deleted using the V2 route, we try to remove both the old `name` identifier and the new `id` identifier in OpenFGA. This is to ensure that we removing the appropriate identifiers in OpenFGA when an alert is deleted both before and after the `alerts_openfga_id` migration script has ran.